rbc/omnisci_backend/omnisci_array.py

```diff
@@ -31,7 +35,101 @@ def buffer_extra_members(self):
 
 
 class Array(Buffer):
-    pass
+    """
+    In HeavyDB, an Array of type `T` is represented as follows:
+
+    .. code-block:: C
+
+        {
+            T* data,
+            int64_t size,
+            int8_t is_null
+        }
+
+    Array is a contiguous block of memory that behaves like a python list.
+
+    Example
+
+
+    .. code-block:: python
+
+        from numba import types
+        from rbc.omnisci_backend import Array
+
+        @omnisci('int64[](int64)')
+        def create_and_fill_array(size):
+            arr = Array(size, types.int64)
+            for i in range(size):
+                a[i] = i
+            return a
+    """
+
+    def __init__(self, size: int, dtype: Union[str, types.Type]) -> None:
+        pass
+
+    def is_null(self) -> bool:
+        pass
+
+    @property
+    def dtype(self):
+        """
+        Data type of the array elements.
+        """
+        pass
+
+    @property
+    def device(self):
+        """
+        ❌ Not implemented
+
+        Hardware device the array data resides on.
+        """
+        pass
+
+    @property
+    def mT(self):
+        """
+        ❌ Not implemented
+
+        Transpose of a amtrix (or a stack of matrices).
```

Comment:
```suggestion
        Transpose of a matrix (or a stack of matrices).
```
What is a matrix in omnisci context? Array of arrays? In any case, since we don't have matrices yet, `mT` should be marked as "Not supported" and mentioning matrix in a doc string does not make sense yet.